public/javascripts/media-optimization-worker.js

```diff
@@ -0,0 +1,165 @@
+function resizeWithAspect(
+  input_width,
+  input_height,
+  target_width,
+  target_height,
+) {
+  if (!target_width && !target_height) {
+    throw Error('Need to specify at least width or height when resizing');
+  }
+
+  if (target_width && target_height) {
+    return { width: target_width, height: target_height };
+  }
+
+  if (!target_width) {
+    return {
+      width: Math.round((input_width / input_height) * target_height),
+      height: target_height,
+    };
+  }
+
+  return {
+    width: target_width,
+    height: Math.round((input_height / input_width) * target_width),
+  };
+}
+
+function logIfDebug(message, debugMode) {
```

Comment:
Minor suggestion: Rather than passing `debugMode` in each time you call this, you could assign a local variable below like this:

```
  let log = settings.debug_mode ? console.log : () => {};
```

Then you can do `log('whatever')` after that and not worry about it.

Reply:
That was the first thing I tried, but the global variable scope inside a dedicated web worker was being weird. Let me try again.